lib/runtime-core/src/cache.rs

```diff
@@ -41,12 +42,13 @@ impl WasmHash {
     /// # Note:
     /// This does no verification that the supplied data
     /// is, in fact, a wasm module.
-    pub fn generate(wasm: &[u8]) -> Self {
+    pub fn generate(wasm: &[u8], backend: Backend) -> Self {
```

Comment:
Can we have this as a non-breaking change. I think near was relying on `generate` and I would like to keep the function signature the same.

Perhaps something like?
```
    pub fn generate_for_backend(wasm: &[u8], backend: Option<Backend>) -> Self {
    }
    pub fn generate(wasm: &[u8]) -> Self {
       Self::generate_for_backend(wasm, Some(Cranelift);
    }
```

Reply:
Sure, the main reason I didn't do that is that it has the potential for bad things to happen, but I'll call it out in a doc comment and call it good enough!